src/utils.jl

```diff
@@ -242,13 +243,21 @@ function read_input(file_name::AbstractString)
         end

```

Comment:
[for later] Github doesn't let me comment in line 49 on the definition of `components(d)`.
to avoid defining the new component market, we may solve this by defining:
``` julia
"Auxiliary function to check if the key 'type' is available in the dictionary d, otherwise false"
has_type(d::AbstractDict) = ("type" in keys(d))
has_type(d) = False  # if d is not an abstract dictionary, then return false

function components(d)
      return Dict(k=>v for (k,v) in d if has_type(v))
end
```

By doing this, all entries in users that do not have the field 'type' would not be considered as components and so the code should work properly.

Reply:
So we should leave in the YAML instead of 
"market_set: 
      market_type: commercial #type of market
      type: market  #type of market
We should set:
"market_set: commercial". Right?

Reply:
Completed